indico/modules/rb_new/client/js/common/bookings/BookingDetails.jsx

```diff
@@ -294,6 +294,27 @@ class BookingDetails extends React.Component {
         });
     };
 
+    transformToLabel = (type) => {
+        switch (type) {
+            case 'bookings': return {label: Translate.string('Current'), style: 'booking'};
+            case 'cancellations': return {label: Translate.string('Cancelled'), style: 'cancellation'};
+            case 'rejections': return {label: Translate.string('Rejected'), style: 'rejection'};
+            case 'otherBookings': return {label: Translate.string('Other bookings'), style: 'other'};
+            default: return undefined;
+        }
+    };
+
+    getLegendLabels = (availability) => {
+        const legendLabels = [];
+        Object.entries(availability).forEach(([type, occurrences]) => {
+            if (occurrences && Object.keys(occurrences).length > 0) {
+                const label = this.transformToLabel(type);
+                label && !legendLabels.some(lab => _.isEqual(lab, label)) && legendLabels.push(label);
```

Comment:
use a normal `if ()` statement here. this kind of "optimization" is something valid if you are a script that's minifying JS, but not something that should be in original source code ;)